File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
    
       char[] sr = s.toCharArray();
        Map<Character, Integer> mp = new HashMap<>();

        // 1. Fill frequency map
        for (char c : sr) {
            mp.put(c, mp.getOrDefault(c, 0) + 1);
        }

        // 2. Find first character with count == 1
        for (int i = 0; i < sr.length; i++) {
            if (mp.get(sr[i]) == 1) {
                return i;
            }
        }

        return -1;
    }
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
      // C++ HashMap (equivalent to Java's HashMap<Character, Integer>)
        unordered_map<char, int> count;

        // First pass: count frequency of each character
        for (char c : s) {
            count[c]++;
        }

        // Second pass: find the first character with a count of 1
        for (int i = 0; i < s.size(); i++) {
            if (count[s[i]] == 1) {
                return i;
            }
        }

        return -1;
     
    }
};